src/hb-subset-plan.cc

```diff
@@ -68,20 +68,95 @@ _remap_indexes (const hb_set_t *indexes,
 }
 
 #ifndef HB_NO_SUBSET_LAYOUT
+static void _collect_subset_lookups (hb_face_t      *face,
+				     hb_tag_t        table_tag,
+				     const hb_set_t *layout_features_to_retain,
+				     bool            retain_all_features,
+				     hb_set_t       *lookup_indices /* OUT */)
+{
+  if (retain_all_features)
+  {
+    hb_ot_layout_collect_lookups (face,
+				  table_tag,
+				  nullptr,
+				  nullptr,
+				  nullptr,
+				  lookup_indices);
+    return;
+  }
+
+  if (hb_set_is_empty (layout_features_to_retain)) return;
+  unsigned num = layout_features_to_retain->get_population () + 1;
+  hb_tag_t *features = (hb_tag_t *) malloc (num * sizeof (hb_tag_t));
+  if (!features) return;
+
+  unsigned i = 0;
+  for (hb_tag_t f : layout_features_to_retain->iter ())
+    features[i++] = f;
+
+  features[i] = 0;
+  
+  hb_ot_layout_collect_lookups (face,
+				table_tag,
+				nullptr,
+				nullptr,
+				features,
+				lookup_indices);
+
+  free (features);
+}
+
+static void _collect_subset_features (hb_face_t      *face,
+				      hb_tag_t        table_tag,
+				      const hb_set_t *layout_features_to_retain,
+				      bool            retain_all_features,
+				      hb_set_t       *lookup_indices /* OUT */)
```

Comment:
This function and the above are pretty much identical except for the function they call. You should be able to have just one function that takes a function point to the collect function to call.